### DISCUSSION TITLE
Difference in the output when I upload .gmt file

### DISCUSSION THREAD

**Discussion by pushtimehta** [zqfang/GSEApy#225] [Q&A]
Hi, 

I would like to tell you that when I use the command gs_res = gp.enrichr(gene_list=data.gene,  gene_sets='MSigDB_Hallmark_2020', organism='human', outdir=None), I get 5 significant gene set. However, when I input the .gmt file of the same gene set library from MSigDB (https://data.broadinstitute.org/gsea-msigdb/msigdb/release/7.2/h.all.v7.2.symbols.gmt) I get only 2 significant pathways. Can you please explain the reason behind this. 

I have checked the contents of both the gene sets. It is exactly the same except for the order of the genes in the gene set. I seems like it affects the scoring. Can you please explain why and how? 

Please let me know at the earliest.

**Comment by zqfang** [comment]
enrichr webservice uses different background genes, to the local input(GMT).

The implementation details of the code will also be subtly different.
